src/ua/griddynamics/Purchase.java

```diff
@@ -0,0 +1,143 @@
+package ua.griddynamics;
+
+import menu.ShowListMenu;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Purchase implements Serializable {
+
+    private static final String FILE_NAME = "purchases.txt";
+
+    private Balance balance;
+    private List<Product> allProducts;
+
+    public Purchase() {
+        this.balance = new Balance();
+        this.allProducts = new ArrayList<>();
+    }
+
+    public void addIncome(double income) {
+        this.balance.addIncome(income);
+    }
+
+    public String showBalance() {
+        return balance.showBalance();
+    }
+
+    public void reduceBalance(double delta) {
+        balance.reduceBalance(delta);
+    }
+
+    public boolean addProduct(Product product) {
+        if (product != null) {
+            allProducts.add(product);
+            return true;
+        }
+        return false;
+    }
+
+    public double getTotalSumByType(ProductType type) {
+        return allProducts.stream()
+                .filter(p -> p.getType() == type)
+                .mapToDouble(Product::getPrice)
+                .sum();
+    }
+
+    public double getTotalSumOfAllProducts() {
+        return allProducts.stream()
+                .mapToDouble(Product::getPrice)
+                .sum();
+    }
+
+    public String showList(int choice) {
+        StringBuilder sb = new StringBuilder();
+        if (ShowListMenu.getInstance(choice).equals(ShowListMenu.ALL)) {
```

Comment:
Purchase class should not now anything about Menu.
This method should have the following signature:
`public String getDescription(ProductType productType) `